src/controllers/userController.js

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable camelcase */
+import userService from '../services/userservice';
+import Util from '../utils/response';
+import { jwtSignUser } from '../utils/index';
+import { hashPassword } from '../helpers/index';
+
+
+const util = new Util();
+/**
+ * @param { class } provide response to user signup activity.
+ */
+class userController {
+  /**
+ * @param {req} req that contains the req body object.
+ * @param {res} res content to be rendered.
+ * @returns {object} Success or failure response on getting all users
+ */
+  static async getUsers(req, res) {
+    try {
+      const allUsers = await userService.getAllUser();
+      if (allUsers.length > 0) {
+        util.setSuccess(200, 'Users retrieved', allUsers);
+      } else {
+        util.setSuccess(200, 'No Users found');
+      }
+      return util.send(res);
+    } catch (error) {
+      util.setError(400, error);
+      return util.send(res);
+    }
+  }
+
+  /**
+ * @param {req} req that contains the req body object.
+ * @param {res} res content to be rendered.
+ * @returns {object} Success or failure response on adding a specific user
+ */
+  static async addUser(req, res) {
+    const { user, emailToken } = req;
+    try {
+      const hashpassword = await hashPassword(user.password);
+      user.password = hashpassword;
+      const {
+        id, email, firstName, lastName,
+      } = await userService.addUser(user);
+      const token = await jwtSignUser(id);
+      util.setSuccess(201, 'user Added!', {
+        token, emailToken, id, email, firstName, lastName,
+      });
+      return util.send(res);
+    } catch (error) {
+      if (error.original.routine === '_bt_check_unique') {
+        util.setError(400, 'Email already exist');
+        return util.send(res);
+      }
+      util.setError(400, error.message);
+      return util.send(res);
+    }
+  }
+
+  /**
+ * @param {req} req that contains the req body object.
+ * @param {res} res content to be rendered.
+ * @returns {object} Success or failure response on updating a  single user details
+ */
+  static async updatedUser(req, res) {
+    const alteredUser = req.body;
+    const { id } = req.params;
+    if (!Number(id)) {
+      util.setError(400, 'Please input a valid numeric value');
+      return util.send(res);
+    }
+    try {
+      const updateUser = await userService.updateUser(id, alteredUser);
+      if (!updateUser) {
+        util.setError(404, `Cannot find user with the id: ${id}`);
+      } else {
+        util.setSuccess(200, 'User updated', updateUser);
+      }
+      return util.send(res);
+    } catch (error) {
+      util.setError(404, error);
+      return util.send(res);
+    }
+  }
+
+  /**
+ * @param {req} req that contains the req body object.
+ * @param {res} res content to be rendered.
+ * @returns {object} Success or failure response on getting a specific user
+ */
+  static async getAUser(req, res) {
```

Comment:
Why this? Shouldn't you just be creating users?

Reply:
Well i wanted to handle all CRUD operation of user. But nevertheless it has been deleted